src/Transformer/AbstractEnumTransformer.php

```diff
@@ -0,0 +1,43 @@
+<?php
+declare(strict_types=1);
+
+namespace Diaclone\Transformer;
+
+use Diaclone\Resource\ResourceInterface;
+
+abstract class AbstractEnumTransformer extends AbstractTransformer
+{
+    public function getPropertyValue($data, $property)
+    {
+        if (empty($property)) {
+            return $data->getValue();
+        }
+
+        // convert property to camelCase
+        $getter = 'get' . str_replace(' ', '', ucwords(str_replace(['-', '_'], ' ', $property)));
+        if (method_exists($data, $getter)) {
+            $enum = $data->$getter();
+
+            return $enum->getValue();
+        }
+
+        return $data->$property->getValue() ?? null;
+    }
+
+    public function transform(ResourceInterface $resource)
+    {
+        return $this->getPropertyValueFromResource($resource);
+    }
+
+    public function untransform(ResourceInterface $resource)
+    {
+        if (!$data = $resource->getData()) {
+            return null;
+        }
+        $propertyTransform = $this->getEnumClass();
+
+        return $propertyTransform::byValue($data);
```

Comment:
I've noticed that this will throw exception when value is not in the Enum, I'd check with:
```
if ($propertyTransform::has($slugString)) {
      return $propertyTransform::byValue($data);
}
```

Reply:
What should happen with the value if it doesn't exist?

Reply:
Ideally we'd have a validation layer pick up bad value and spit out allowed values. I guess we could pass the original string through but that might break the rest of the flow (where Enum class is expected) if validation layer is missing/doesn't catch. Or you could re-throw a transformation exception instead of Enum's `InvalidArgumentException`.

Reply:
I'll throw an exception. I don't like the idea of passing a scalar on through.